test/src/xquery/arrays/arrays.xql

```diff
@@ -134,6 +134,14 @@ function arr:function-item-invalid() {
         $a("x")
 };
 
+declare
```

Comment:
what about testing [0] [-1] [1] and [3] ? these are cornercases...

Reply:
I have now incorporated your further tests.

Reply:
but the logic is still not throwing the correct errors, right?

Reply:
I added further tests. The logic was correct, but the error code was wrong, I have corrected that now.